src/shared/tooltip.rs

```diff
@@ -41,22 +55,109 @@ live_design! {
 
 #[derive(Live, LiveHook, Widget)]
 pub struct Tooltip {
-    #[deref]
-    view: View,
+    #[rust]
+    opened: bool,
+
+    #[live]
+    #[find]
+    content: View,
+
+    #[redraw]
+    #[rust(DrawList2d::new(cx))]
+    draw_list: DrawList2d,
+
+    #[live]
+    draw_bg: DrawQuad,
+    #[layout]
+    layout: Layout,
+    #[walk]
+    walk: Walk,
 }
 
 impl Widget for Tooltip {
     fn handle_event(&mut self, cx: &mut Cx, event: &Event, scope: &mut Scope) {
-        self.view.handle_event(cx, event, scope);
+        self.content.handle_event(cx, event, scope);
+    }
+
+    fn draw_walk(&mut self, cx: &mut Cx2d, scope: &mut Scope, _walk: Walk) -> DrawStep {
+        self.draw_list.begin_overlay_reuse(cx);
+
+        cx.begin_pass_sized_turtle(self.layout);
+        self.draw_bg.begin(cx, self.walk, self.layout);
+
+        if self.opened {
+            let _ = self.content.draw_all(cx, scope);
+        }
+
+        self.draw_bg.end(cx);
+
+        cx.end_pass_sized_turtle();
+        self.draw_list.end(cx);
+
+        DrawStep::done()
+    }
+
+    fn set_text(&mut self, text: &str) {
+        self.label(id!(tooltip_label)).set_text(text);
+    }
+}
+
+impl Tooltip {
+    pub fn set_pos(&mut self, cx: &mut Cx, pos: DVec2) {
+        self.apply_over(
+            cx,
+            live! {
+                content: { margin: { left: (pos.x), top: (pos.y) } }
+            },
+        );
+    }
+
+    pub fn show(&mut self, cx: &mut Cx) {
+        self.opened = true;
+        self.redraw(cx);
+    }
+
+    pub fn show_with_options(&mut self, cx: &mut Cx, pos: DVec2, text: &str) {
+        self.opened = true;
+        self.set_text(text);
+        self.set_pos(cx, pos);
+        self.redraw(cx);
```

Comment:
Nitpick: Any reason to not use `show(cx)` here? If we use it here, then we will know it works and will not count as deadcode in moxin 😉

```suggestion
        self.set_text(text);
        self.set_pos(cx, pos);
        self.show(cx);
```